### DISCUSSION TITLE
Cannot download image; how do I set the proxy?

### DISCUSSION THREAD

**Discussion by mwidjaja1** [lima-vm/lima#1618] [General]
Hey gang 👋🏼! I'm trying to run `limactl start` but when I do so, I get a `failed to download <insert OS image here>` type error where it says `read: connection reset by peer`.

This isn't too surprising because I'm required to run behind an http/https proxy that is set in my terminal with an env `$HTTP_PROXY` and `$HTTPS_PROXY` variable. My question is, how do I get this proxy information to lima so I can get these images installed?

Thanks!

**Comment by afbjorklund** [comment]
The environment variables are called `http_proxy` and `HTTPS_PROXY`, confusingly enough.

EDIT: it should accept either case of the variables, besides the historical ones

**Reply by mwidjaja1** [reply]
Thanks but I have variables like that too, in all lowercase. Do you know where they should be added because Lima is definitely not picking them up when they’re just added to my Bash’s Environment.

**Reply by afbjorklund** [reply]
I think they go in `env` for the VM, or otherwise in the login scripts or container run

https://github.com/lima-vm/lima/blob/v0.16.0/examples/default.yaml#L383_L390

**Reply by mwidjaja1** [reply]
Hmm, unfortunately, I don't think that worked. I think my issue is that the core of the problem is not the proxy inside of my container, but rather the actual Go Downloader at

https://github.com/lima-vm/lima/blob/9d29da396c164fea261bc547b4537d470aa3a824/pkg/downloader/downloader.go

This downloader, which downloads my images, doesn't appear to have any place for me to inject my proxy. Without it, Lima is unable to download the image required to initialize my container. Would you have an idea where this guy gets its network info from?

Thanks!

**Reply by afbjorklund** [reply]
~~Maybe it needs a special call, there seems to be some net/http `ProxyFromEnvironment` that could be relevant to this issue.~~

Can you open a GitHub issue for it ?

**Reply by afbjorklund** [reply]
Did not see any new issues, not sure if the problem is still there or if code is still needed.

**Reply by afbjorklund** [reply]
`HTTPS_PROXY=http://localhost:3128 limactl start template://debian` has no issues contacting squid

so the default http.Transport should already be using the proxy variables from the environment...

**Comment by Govindaraju** [comment]
I m facing the same issue whilst starting colima. Here are the details. Please could you suggest the way around it ?

colima version 0.5.5
limactl version 0.16.0

Actual error message
failed to download "https://github.com/abiosoft/alpine-lima/releases/download/colima-v0.5.5/alpine-lima-clm-3.18.0-aarch64.iso": Get "https://github.com/abiosoft/alpine-lima/releases/download/colima-v0.5.5/alpine-lima-clm-3.18.0-aarch64.iso": dial tcp: lookup github.com: no such host

We are using a web proxy as well.

I tried injecting them through environment variables.

**Reply by afbjorklund** [reply]
Sounds like an issue for colima

**Comment by Govindaraju** [comment]
Hi

Looks like my proxy password had % as one of the special characters and
colima code didnt like it. Can you please check that once ?

Regards
Govind

**Reply by jandubois** [reply]
Try replacing the `%` with `%25` (so `x%y` would become `x%25y`).

**Reply by jandubois** [reply]
If that doesn't fix it, maybe report your issue on the colima repo instead?
